gluster/cli/peer.py

```diff
@@ -47,6 +47,34 @@ def detach(host):
     cmd = ["detach", host]
     return peer_execute(cmd)
 
+def detachAll():
+    """
+    Removes All Hosts from Cluster
+
+    :returns: nothing
+     print: success and failure messages, raises
+    """
+    try:
+        peers = parse_peer_status(peer_execute_xml(["status"]))
+    except:
+        print "Error in fetching peer status"
+    if len(peers) == 0:
+        print "No peers to Detach"
+    else:
+        print "Found "+str(len(peers))+" peers"
+        for peer in peers:
+                host = peer["hostname"]
+                if peer["connected"] == "Connected":
+                    cmd = ["detach",host]
+                    try:
+                        result = peer_execute(cmd)
+                        out = str(host)+" "+result
+                        print out
+                    except:
+                        print "Error in detaching"
```

Comment:
Collect the failure error messages here. At the end of for loop you can raise exception if errors list is not empty.